staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -111,7 +111,9 @@ func (e *eventBroadcasterImpl) refreshExistingEventSeries() {
 	for isomorphicKey, event := range e.eventCache {
 		if event.Series != nil {
 			if recordedEvent, retry := recordEvent(e.sink, event); !retry {
-				e.eventCache[isomorphicKey] = recordedEvent
+				if recordedEvent != nil {
```

Comment:
Please add a unit test for it.

Reply:
Added a test that simulates an apiserver error